adapters/appnexus/appnexus.go

```diff
@@ -285,6 +286,15 @@ func handleLegacyParams(appnexusExt *openrtb_ext.ExtImpAppnexus) {
 	}
 }
 
+func validateAppnexusExt(appnexusExt *openrtb_ext.ExtImpAppnexus) error {
+	if appnexusExt.PlacementId == 0 && (appnexusExt.InvCode == "" || appnexusExt.Member == "") {
+		return &errortypes.BadInput{
+			Message: "No placement or member+invcode provided",
+		}
+	}
+	return nil
+}
```

Comment:
Perhaps this is outside of the scope of this PR given this logic is a rewrite but why do we need this function? Isn't this validation already performed upstream during bidder param validation?
```
"oneOf": [{
    "oneOf": [{
      "required": ["placementId"]
    }, {
      "required": ["placement_id"]
    }]
  }, {
    "oneOf": [{
      "required": ["invCode", "member"]
    }, {
      "required": ["inv_code", "member"]
    }]
  }],
```

Are we keeping this logic just so we can be sure `handleLegacyParams` doesn't perform a copy that results in an invalid request?

Reply:
It is to be consistent with Prebid.JS. It is always difficult to get prebid.JS updated in production, and you can't do A/B testing if the js and server adapter parameters are different.